core/src/main/scala/org/broadinstitute/dsde/rawls/model/SpendReportingModel.scala

```diff
@@ -22,6 +25,53 @@ case class SpendReportingAggregationKeyWithSub(key: SpendReportingAggregationKey
 )
 
 case class SpendReportingResults(spendDetails: Seq[SpendReportingAggregation], spendSummary: SpendReportingForDateRange)
+
+object SpendReportingResultsConvertor {
+  def apply(spendReport: SpendReport): SpendReportingResults = {
+
+    def mapSpendReportingForDateRange(
+      spendReportingForDateRange: bio.terra.profile.model.SpendReportingForDateRange
+    ): SpendReportingForDateRange =
+      SpendReportingForDateRange(
+        spendReportingForDateRange.getCost,
+        spendReportingForDateRange.getCredits,
+        spendReportingForDateRange.getCurrency,
+        Option.empty,
+        Option.empty,
+        category = Option.apply(TerraSpendCategories.withName(spendReportingForDateRange.getCategory.toString))
+      )
+
+    def mapSpendReportingAggregation(
+      spendReportingAggregation: bio.terra.profile.model.SpendReportingAggregation
+    ): SpendReportingAggregation = {
+
+      val spendData = spendReportingAggregation.getSpendData.asScala
+        .map(mapSpendReportingForDateRange)
+        .toList
+
+      SpendReportingAggregation(
+        SpendReportingAggregationKeys.withName(spendReportingAggregation.getAggregationKey.name()),
+        spendData
+      )
+    }
+
+    val spendDetails: Seq[SpendReportingAggregation] =
+      spendReport.getSpendDetails.asScala
+        .map(mapSpendReportingAggregation)
+        .toList
+
+    val spendSummary = SpendReportingForDateRange(
+      spendReport.getSpendSummary.getCost,
+      spendReport.getSpendSummary.getCredits,
+      spendReport.getSpendSummary.getCurrency,
+      Option.apply(DateTime.parse(spendReport.getSpendSummary.getStartTime)),
+      Option.apply(DateTime.parse(spendReport.getSpendSummary.getEndTime))
+    )
+
+    SpendReportingResults.apply(spendDetails, spendSummary)
+  }
+}
+
```

Comment:
Agree. This will be more idiomatic solution in this case. I initially start using this approach per @blakery recommendation, but by some reason it didn't work. Most likely I did something wrong. So, I unblocked myself and used another approach.

Reply:
Apply should belong to the corresponding companion object.